src/storage/page_buffer.c

```diff
@@ -7425,6 +7431,12 @@ pgbuf_lock_page (THREAD_ENTRY * const thread_p, const VPID * const vpid, PAGE_FE
     }
   assert ((pgbuf_lock_state == PGBUF_LOCK_WAITER) != (cur_buffer_lock == nullptr));	// xor
 
+  if (fetch_mode == OLD_PAGE_IF_IN_BUFFER_OR_IN_TRANSIT)
+    {
+      return PGBUF_LOCK_NOT_NEEDED;
+    }
```

Comment:
If `pgbuf_lock_state == PGBUF_LOCK_WAITER`, it means the page was loaded to page buffer and must be fixed.

If at this point `pgbuf_lock_state == PGBUF_LOCK_HOLDER` and `fetch_mode == OLD_PAGE_IF_IN_BUFFER_OR_IN_TRANSIT`, then page does not have to be loaded to page buffer and lock may be changed to `PGBUF_LOCK_NOT_NEEDED`.

Reply:
Done. My bad.